omnibus.rb

```diff
@@ -44,9 +44,14 @@
 fetcher_retries 3
 fetcher_read_timeout 120
 
-# We limit this to 10 workers to eliminate transient timing issues in the
-# way Ruby (and other components) compiles on some more esoteric *nixes.
-workers 10
+# Work around for solaris zones, provide a less-bad number of workers
+if solaris? && Ohai['cpu']['total'] == 0
+  workers 10
+end
+
+# Some platforms do not have a UTF-8 locale, so we need to enforce one
+# or else the cacert chain will break among other things
+Encoding.default_external = Encoding::UTF_8
```

Comment:
Do you actually have a requirement to have ruby use some other encoding as the default external encoding though?

AFAIK you can force this everywhere and it should be perfectly fine.  You don't need UTF_8 locales installed on the server (as this is not actually a locale, this is just charset support, there's no 'en_US' part here).  Ruby always supports this, we can and should always set it.  If we just do it in the binary and forget about making it a config option we can solve it forever with a one-line patch with minimal effort, and not outsmart ourselves by considering edge cases that are unlikely to ever happen.